### PR TITLE
SNOW-829454: Async API Example

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
### Description
Prepare Async API example.

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by sfc-gh-pbulawa** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cmd/async/async.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	}
+	defer db.Close()
+
+	var rows driver.Rows
+	fmt.Println("Lets simulate long running query")
+	rows, err = runAsyncQuery(db, "CALL SYSTEM$WAIT(10, 'SECONDS')")
+	fmt.Println("The query is running asynchronously")
+	printResult(rows)
+
+	fmt.Println("Lets simulate long running query using the second way")
```
Comment (sfc-gh-pfus) [inline cmd/async/async.go:79]: Can we describe briefly this "second way" instead of printing non meaningful message?

File: cmd/async/async.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+	db, err := sql.Open("snowflake", dsn)
+	if err != nil {
+		log.Fatalf("failed to connect. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	var rows driver.Rows
+	fmt.Println("Lets simulate long running query")
+	rows, err = runAsyncQuery(db, "CALL SYSTEM$WAIT(10, 'SECONDS')")
```
Comment (sfc-gh-dheyman-1) [inline cmd/async/async.go:75]: I don't think this is the best example. Perhaps we should pass there a callback function that would be invoked when results are eventually fetched?

File: cmd/async/async.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	"fmt"
+	sf "github.com/snowflakedb/gosnowflake"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func getDSN() (string, *sf.Config, error) {
```
Comment (sfc-gh-igarish) [inline cmd/async/async.go:17]: I have seen similar code in other examples, is it possible to make a util func to do same/similar stuff?

Comment (sfc-gh-pbulawa) [reply cmd/async/async.go:17]: I believe that this should be in the scope of another issue as we have 11 duplicates of the function thus I have created a new issue for that: SNOW-832816.

File: cmd/async/async.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"flag"
+	"fmt"
+	sf "github.com/snowflakedb/gosnowflake"
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply cmd/async/async.go:9]: Done

File: cmd/async/async.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		log.Fatalf("failed to create DSN from Config: %v, err: %v", cfg, err)
+	}
+
+	db, err := sql.Open("snowflake", dsn)
+	if err != nil {
+		log.Fatalf("failed to connect. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	var rows driver.Rows
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply cmd/async/async.go]: Done

File: cmd/async/async.go (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply cmd/async/async.go]: Described both ways and named methods to be more specific.

**Review by sfc-gh-pfus — APPROVED** [review]

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**